src/parser/jobs/mnk/modules/PerfectBalance.tsx

```diff
@@ -27,36 +32,114 @@ const PB_BAD_ACTIONS: ActionKey[] = [
 
 interface Balance {
 	bads: number
-	stacks: number
+	stacks: number // this is the number of stacks the window began with (may be less than the max for dungeons/24mans with a carried-over PB)
 	used: number
+	start: number
 }
 
-export class PerfectBalance extends Analyser {
+const GAUGE_FADE = 0.25
+const OPO_GAUGE_COLOR = Color('#a256dc')
+const RAPTOR_GAUGE_COLOR = Color('#57b39a')
+const COEURL_GAUGE_COLOR = Color('#d7548e')
+const LUNAR_NADI_COLOR = Color('#8a57c4')
+const SOLAR_NADI_COLOR = Color('#e7dbd3')
+
+export class PerfectBalance extends Gauge {
 	static override debug = false
 	static override handle = 'perfectBalance'
 	static override title = t('mnk.pb.title')`Perfect Balance`
 	static override displayOrder = DISPLAY_ORDER.PERFECT_BALANCE
 
-	@dependency private data!: Data
 	@dependency private suggestions!: Suggestions
 
 	private badActions: Array<Action['id']> = []
 	private formActions: Array<Action['id']> = []
+	private opoActions: Array<Action['id']> = []
+	private raptorActions: Array<Action['id']> = []
+	private coeurlActions: Array<Action['id']> = []
+	private blitzActions: Array<Action['id']> = []
 
 	private current: Balance | undefined
 	private history: Balance[] = []
 
 	private perfectHook?: EventHook<Events['action']>
+	private beastTimeoutHook?: TimestampHook
+
+	private opoBeastGauge = this.add(new CounterGauge({
+		maximum: 3,
+		initialValue: 0,
+		graph: {
+			handle: BEAST_GAUGE_HANDLE,
+			label: <Trans id="mnk.gauge.resource.beast.opo">Opo-opo</Trans>,
+			color: OPO_GAUGE_COLOR.fade(GAUGE_FADE),
+		},
+	}))
+	private raptorBeastGauge = this.add(new CounterGauge({
+		maximum: 3,
+		initialValue: 0,
+		graph: {
+			handle: BEAST_GAUGE_HANDLE,
+			label: <Trans id="mnk.gauge.resource.beast.raptor">Raptor</Trans>,
+			color: RAPTOR_GAUGE_COLOR.fade(GAUGE_FADE),
+		},
+	}))
+	private coeurlBeastGauge = this.add(new CounterGauge({
+		maximum: 3,
+		initialValue: 0,
+		graph: {
+			handle: BEAST_GAUGE_HANDLE,
+			label: <Trans id="mnk.gauge.resource.beast.coeurl">Coeurl</Trans>,
+			color: COEURL_GAUGE_COLOR.fade(GAUGE_FADE),
+		},
+	}))
+	private lunarNadiGauge = this.add(new CounterGauge({
+		maximum: 1,
+		initialValue: 0,
+		graph: {
+			handle: NADI_GAUGE_HANDLE,
+			label: <Trans id="mnk.gauge.resource.nadi.lunar">Lunar Nadi</Trans>,
+			color: LUNAR_NADI_COLOR.fade(GAUGE_FADE),
+		},
+	}))
+	private solarNadiGauge = this.add(new CounterGauge({
+		maximum: 1,
+		initialValue: 0,
+		graph: {
+			handle: NADI_GAUGE_HANDLE,
+			label: <Trans id="mnk.gauge.resource.nadi.solar">Solar Nadi</Trans>,
+			color: SOLAR_NADI_COLOR.fade(GAUGE_FADE),
+		},
+	}))
```

Comment:
These should be a boolean gauge, or history of bool states, not counters.

Reply:
BooleanGauge doesn't exist today, and it's not *wrong* per se, so they'll stay as they are. I have thoughts about how to polish both the nadis and the Beast gauge even further but that will be a separate future PR.

Reply:
The only other "boolean" gauge in the game I can think of is BLM's paradox [which we also currently model with a CounterGauge](https://github.com/xivanalysis/xivanalysis/blob/626d6406f5ca35a8f824d0909e56e424280202a6/src/parser/jobs/blm/modules/Gauge.tsx#L148). Since it's a relatively large undertaking to create a new gauge type and it has such limited applicability, I don't think it's worth blocking on.

Reply:
This is handled in #1868